tools/pindel/pindel.py

```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env python
+
+import logging
+import argparse, os, shutil, subprocess, sys, tempfile, time, shlex
+from multiprocessing import Pool
+import vcf
+
+def execute(cmd, output=None):
+    import subprocess, sys, shlex
+    # function to execute a cmd and report if an error occur
+    print(cmd)
+    try:
+        process = subprocess.Popen(args=shlex.split(cmd), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout,stderr = process.communicate()
+    except Exception, e: # une erreur de ma commande : stderr
+        sys.stderr.write("problem doing : %s\n%s\n" %(cmd, e))
+        return
+    if output:
+        output = open(output, 'w')
+        output.write(stdout)
+        output.close()
+    if stderr != '': # une erreur interne au programme : stdout (sinon, souvent des warning arrete les programmes)
+        sys.stdout.write("warning or error while doing : %s\n-----\n%s-----\n\n" %(cmd, stderr))
+
+
+def indexBam(workdir, inputFastaFile, inputBamFile, bam_number, inputBamFileIndex=None):
+    inputFastaLink = os.path.join(os.path.abspath(workdir), "reference.fa" )
+    if not os.path.exists(inputFastaLink):
+        os.symlink(inputFastaFile, inputFastaLink)
+        cmd = "samtools faidx %s" %(inputFastaLink)
+        execute(cmd)
+    inputBamLink = os.path.join(os.path.abspath(workdir), "sample_%d.bam" % (bam_number) )
+    if os.path.isfile(inputBamLink):  ### JHL: Check to ensure file does not already exist, leading to wrapper error.
+        os.remove(inputBamLink)
+    os.symlink(inputBamFile, inputBamLink)
+    if inputBamFileIndex is None:
+        cmd = "samtools index %s" %(inputBamLink)
+        execute(cmd)
+    else:
+        if os.path.isfile(inputBamLink + ".bai"):
+            os.remove(inputBamLink + ".bai")
+        os.symlink(inputBamFileIndex, inputBamLink + ".bai")
+    return inputFastaLink, inputBamLink
```

Comment:
Since we downloaded most of these wrappers from Galaxy toolshed, we have from the beginning only been testing the functionality with all the code embedded in the wrapper. But I agree, for maintainability, it would be good to have separate modules of utility code and import them.